src/org/zaproxy/zap/extension/ascanrulesBeta/InsecureHTTPMethod.java

```diff
@@ -451,5 +500,67 @@ private boolean testConnect (HttpMessage baseMsg, String thirdpartyHost, int thi
 		}
 		return false;
 	}
+	
+	private final MessageAndEvidence testHttpMethod(String httpMethod) throws IOException{
+		
+		httpMethod = httpMethod.toLowerCase().trim();
+		
+		if (httpMethod.equals("track")||httpMethod.equals("trace")||httpMethod.equals("connect")) {
+			/* We have separate methods for them */
+			return null;
+		}
+		
+        final HttpRequestHeader traceRequestHeader = this.getBaseMsg().getRequestHeader();
+        
+        traceRequestHeader.setMethod(httpMethod);
+        traceRequestHeader.setVersion(HttpRequestHeader.HTTP11);
+ 
+        final HttpMessage tracemsg = getNewMsg();
+        tracemsg.setRequestHeader(traceRequestHeader);
+        sendAndReceive(tracemsg, true); // do not follow redirects. That might ruin our day.
+        MessageAndEvidence ret = null;
+ 
+        final int responseCode = tracemsg.getResponseHeader().getStatusCode();
+ 
+        if (log.isDebugEnabled()) {
+        	log.debug("Request Method: " + httpMethod);
+        	log.debug("Response Code: " + responseCode);
+        }
+ 
+        if (responseCode == 200) {
```

Comment:
Also is further logic needed here? What if the response is 401/403? (Wouldn't that mean it is enabled, your account/user just can't use it ... raise alert at reduced Risk?)